cmd/neofs-cli/internal/client/client.go

```diff
@@ -336,28 +403,54 @@ type DeleteObjectPrm struct {
 
 // DeleteObjectRes groups resulting values of DeleteObject operation.
 type DeleteObjectRes struct {
-	cliRes *client.ObjectDeleteRes
+	addrTombstone *addressSDK.Address
 }
 
 // TombstoneAddress returns address of the created object with tombstone.
 func (x DeleteObjectRes) TombstoneAddress() *addressSDK.Address {
-	return x.cliRes.TombstoneAddress()
+	return x.addrTombstone
 }
 
 // DeleteObject marks object to be removed from NeoFS through tombstone placement.
 //
 // Returns any error prevented the operation from completing correctly in error return.
-func DeleteObject(prm DeleteObjectPrm) (res DeleteObjectRes, err error) {
-	var delPrm client.DeleteObjectParams
+func DeleteObject(prm DeleteObjectPrm) (*DeleteObjectRes, error) {
+	var delPrm client.PrmObjectDelete
 
-	delPrm.WithAddress(prm.objAddr)
+	if id := prm.objAddr.ContainerID(); id != nil {
+		delPrm.FromContainer(*id)
```

Comment:
As a side-note, why do we use non-pointer parameters in SDK? It seems like we mostly use pointers in other places.

Reply:
Using pointers in parameter structures is inappropriate for two reasons:

1. we never modify them, so we need to mention this fact in package docs clearly;
2. pointer binding adds garbage collection overhead, which I would like to avoid if possible.

But u r right, this approach differs with other places in project. In the future, all such places will be reworked according to the principle "RO - value, RW - pointer". Also, for example, I want to get away from pointer slices.

Reply:
I wouldn't worry about garbage collection ovehead, as here, for example the object already exists, so it will be collected anyway. In addition the (possible) overhead is much less than network-related delays but code complexity increase is much higher IMO.

Reply:
> It is also not obvious whether copy overhead is less than that of a garbage collection.

I agree, but we shouldn't try to pseudo-optimize prevent stack growth with passing pointers where it is not necessary (like in parameters setters). https://github.com/golang/go/wiki/CodeReviewComments#pass-values